src/libutil/file-system.cc

```diff
@@ -21,36 +21,43 @@ namespace fs = std::filesystem;
 
 namespace nix {
 
-Path absPath(PathView path, std::optional<PathView> dir, bool resolveSymlinks)
+#ifdef __GNUC__
+std::unique_ptr<char, FreeDeleter> getCwd()
 {
-    std::string scratch;
+    return std::unique_ptr<char, FreeDeleter> { getcwd(NULL, 0) };
```

Comment:
is this called so often that it must be really fast and small, so that it is worth spilling this memory management over the code that wants the cwd string?

If no, then please create a function like `std::string getCwd()` that copies the content and then frees the original string. this way we don't need anyone to know about custom deleters for this purpose.